synapse/replication/http/send_event.py

```diff
@@ -106,7 +106,7 @@ def on_POST(self, request):
             event.event_id, event.room_id,
         )
 
-        yield self.event_creation_handler.handle_new_client_event(
+        yield self.event_creation_handler.persist_and_notify_client_event(
```

Comment:
do we not need to remove the push actions from the staging area if this fails?

Reply:
We remove push actions in a try finally block after we call `handle_push_actions_for_event` in `handle_new_client_event`

Reply:
Ah yes. I was forgetting that the call to the master was synchronous and that the error would propagate back to the worker, which will do the cleanup.